articles/introduction-to-built-in-data-structures-in-java/index.md

```diff
@@ -0,0 +1,304 @@
+
+```java
+class Node { // Node class containing a single pointer and data
+  Node next; // 'next' is used as pointer
+  int data; // 'data' is used for storing values in the node
+  // Constructor for initializing the variables
+  Node() { 
+    next = null;
+    data = 0;
+  }
+}
+```
+
+In the `java.util` package, we simplify the tasks of manual creation for memory handling, by using `LinkedList` data structure. It contains pre-defined methods for building a linked list data structure.
+
+Syntax for creating `LinkedList` data structure:
+
+```java
+import java.util.LinkedList; // Import package
+
+LinkedList<String> list2 = new LinkedList<String>(); // Create new object
+```
+
+##### Add Elements
+
+In the user-defined linked list data structure, for adding new data, we have to manually move the pointers and assign data. Whereas, in the `LinkedList` data structure, we make use of `add(Object)` or `add(int index, Object)` methods for adding new elements.
+
+Example:
+
+```java
+list2.add("Hello"); // Add "Hello" to the LinkedList list2
+// OUTPUT: ["Hello"]
+list2.add("World"); // Add "World" to the LinkedList list2
+// OUTPUT: ["Hello", "World"]
+list2.add(1, "Computer"); // Add "Computer" to the LinkedList list2 at index 1
+// OUTPUT: ["Hello", "Computer", "World"]
+```
+
+##### Update Elements
+
+To update elements in the user-defined linked list, we have to traverse the whole linked list from the "HEAD" pointer to update the particular data.
+
+`LinkedList` data structure abstracts the whole process with the `set(index, data)` method.
+
+Example:
+
+```java
+list2.set(0, "Welcome"); // Update the string at index 0 to "Welcome"
+// OUTPUT: ["Welcome", "Computer", "World"]
+```
+
+##### Remove Elements
+
+Similarly, for deletion of an element in a user-defined linked list is cumbersome, where the pointer of the previous element is pointed to the address of the next element. When working with huge data, working with pointers is not easy.
+
+`LinkedList` data structure uses `remove(Object)` methods for deletion of elements from the linked list.
+
+Example:
+
+```java
+list2.remove(1); // Removes "Computer" from list2
+// OUTPUT: ["Welcome", "World"]
+```
+
+#### HashMap
+
+A Map is a data structure that helps us in mapping Key and Value pairs. Similarly, in Java, we make use of `HashMap`, which hashes the Key by an index. By hashing, we will be able to access the Value by specifying the index.
+
+Syntax for creating a `HashMap`:
+
+```java
+import java.util.HashMap; // Importing package
+
+HashMap<Integer, String> map1 = new HashMap<Integer, String>(); // Creating a HashMap object
+```
+
+##### Adding Elements
+
+`HashMap` is one of the important data structures, where mapping the key and the value helps in accessing the elements with very little time complexity. To map a key with a value, we make use of the `put(key, value)` method.
```

Comment:
> accessing the elements with very little time complexity.

It would be great if you can specify the exact average case time complexity of accessing an element in a HashMap.